rfc/0052-sdk3-full-text-search.md

```diff
@@ -864,6 +874,21 @@ If the user has specified a vector search, the SDK should raise a `FeatureNotAva
 
 If not, the SDK needs to allow the operation.  The `SearchQuery` inside the `SearchRequest` can easily be forwarded to the existing couchbase2 implementation for `cluster/scope.searchQuery()`. 
 
+### FeatureNotAvailable handling
+
+Iff a `VectorSearch` is included, before sending anything to the server, the SDK will check for the presence of this cluster capability which will only be returned by clusters that are fully upgraded to 7.6.0 or above:
+```
+  "clusterCapabilities": {
+   "search": ["vectorSearch"]
+  }
+```
+If it is not present the SDK will raise `FeatureNotAvailableException` with a message along the lines of "Vector queries are available from Couchbase Server 7.6.0 and above".
+
+The SDK will check for this capability in the most recent config it received.
+If it does not currently have one (in which case one should already be in the process of being asynchronously fetched), the SDK will wait for it to be available.
```

Comment:
I feel like this adds edge cases and potential performance problems for not much benefit.

If there is no config, chances are nothing is bootstrapped yet and the whole thing will fail anyways.